### DISCUSSION TITLE
Support Polymorphic relationships in Join fields

### DISCUSSION THREAD

**Discussion by tyteen4a03** [payloadcms/payload#8861] [Feature Requests & Ideas]
It would be great if we can support polymorphic relationships in Join fields. The initial version can be limited to a single "relationTo".

**Comment by DanRibbens** [comment]
Hey @tyteen4a03, 

Can you give me an example of the collections and relationships that you have where you want to have a join to a polymorphic field. I'm having a hard time wrapping my head around a real world use-case.

**Reply by tyteen4a03** [reply]
I am developing an inventory system for a game. I have an InventoryItem entity which can be owned by either Players, or Clubs.

Currently, it's not possible to put a Join field on the Player on what InventoryItem the player owns.

**Reply by smoothdvd** [reply]
+1
Like Polymorphic Associations in Rails. https://guides.rubyonrails.org/association_basics.html#polymorphic-associations

**Reply by n-a-t-j-b** [reply]
I have another example... my Author field (as distinct from my createdBy/lastUpdatedBy userids) can be either a Partner, Community or a User - it's used to show who the content (e.g. articles, videos, books, etc.) is contributed by/attributed to.  The author is also used in the page path (e.g. /articles/user1/article1, /articles/community1/article1).

As such, this becomes very helpful for landing pages showcasing different types of content contributed by a single author (irrespective of their author type) e.g.

/users/user1 (join shows articles, books, videos etc. contributed by this user)
/partners/partner1 (join shows articles, books, videos etc. contributed by this partner)
/communities/community1 (join shows articles, books, videos etc. contributed by this community)

Hope this makes sense and adds to the value of joins on polymorphic join fields.

p.s.  Thanks for all the awesome work to date!

**Comment by DanRibbens** [comment]
With https://github.com/payloadcms/payload/pull/9990 I'm working on supporting polymorphic relationship fields.

This is different than the join being able to pull in many different relationships in one array.

I hope this helps.
